server/src/main/java/org/elasticsearch/search/aggregations/InternalMultiBucketAggregation.java

```diff
@@ -73,16 +73,30 @@ protected InternalMultiBucketAggregation(StreamInput in) throws IOException {
     public Object getProperty(List<String> path) {
         if (path.isEmpty()) {
             return this;
-        } else if (path.get(0).equals("_bucket_count")) {
+        }
+        String aggName = path.get(0);
+        if (aggName.equals("_bucket_count")) {
             return getBuckets().size();
-        } else {
-            List<? extends InternalBucket> buckets = getBuckets();
-            Object[] propertyArray = new Object[buckets.size()];
-            for (int i = 0; i < buckets.size(); i++) {
-                propertyArray[i] = buckets.get(i).getProperty(getName(), path);
+        }
+        List<? extends InternalBucket> buckets = getBuckets();
+
+        // This is a bucket key, look through our buckets and see if we can find a match
```

Comment:
should we have more tests around invalid input to this? looks to me like this could fail in unintentional ways?

Reply:
++ it does seem fragile.  I'll see about bulking up tests around this parsing in general.

I actually went down the rabbit hole of removing these `List<String>` paths and using the typed `AggregationPath` object (which removes the need to check for apostrophes, etc) but it ended up becoming very complicated and BWC-breaking because we expose these methods directly to clients (including the HLRC) :(